pyresttest/validators.py

```diff
@@ -289,6 +290,24 @@ def parse(cls, config, extractor_base=None):
         return cls.configure_base(config, base)
 
 
+class CookieExtractor(AbstractExtractor):
+    """ Extractor that pulls out a named cookie """
+    extractor_type = 'cookie'
+    is_header_extractor = True
+
+    def extract_internal(self, query = None, args = None, body = None, headers = None):
+        extracted = [y[1] for y in filter(lambda x: x[0] == 'set-cookie', headers) if y[1].startswith(query + '=')]
+        if len(extracted) == 0:
+            raise ValueError("Cookie named {0} not known".format(query))
+        else:
+            return str(dict(Cookie.SimpleCookie(extracted[0]))[query].value)
```

Comment:
Please, can you explain what is happening here?   It feels like some of the casting/here may not be needed.   Also, how will this handle if the same cookie is set multiple times?  (Which IIRC is legal in HTTP, if probably unlikely.)